lua/plenary/curl.lua

```diff
@@ -37,6 +37,15 @@ local F = require "plenary.functional"
 local J = require "plenary.job"
 local P = require "plenary.path"
 
+
+local flatten = function(t)
+  if vim.fn.has "nvim-0.11" == 1 then
+    return vim.iter(t):flatten():totable()
```

Comment:
```suggestion
    return vim.iter(t):flatten(math.huge):totable()
```

Reply:
Do you think this is relevant for both flatten compat implementations?

Reply:
Additionally, all the tests seem to pass as is. If it necessary to increase the depth, should we add a test for this case to remind people in future as I think the vim.iter space is due more changes